core/rootvalue.go

```diff
@@ -573,116 +526,168 @@ func (root *RootValue) PutTypes(ctx context.Context, typ *typecollection.TypeCol
 	if typ == nil {
 		return root, nil
 	}
-	data, err := typ.Serialize(ctx)
-	if err != nil {
-		return nil, err
-	}
-	dataBlob, err := types.NewBlob(ctx, root.vrw, bytes.NewReader(data))
+	newRoot, err := rootobject.PutCollection(ctx, root, typ)
 	if err != nil {
 		return nil, err
 	}
-	ref, err := root.vrw.WriteValue(ctx, dataBlob)
-	if err != nil {
-		return nil, err
-	}
-	newStorage, err := root.st.SetTypes(ctx, ref.TargetHash())
-	if err != nil {
-		return nil, err
-	}
-	return root.withStorage(newStorage), nil
+	return newRoot.(*RootValue), nil
 }
 
 // RemoveTables implements the interface doltdb.RootValue.
 func (root *RootValue) RemoveTables(
 	ctx context.Context,
 	skipFKHandling bool,
 	allowDroppingFKReferenced bool,
-	tables ...doltdb.TableName,
+	originalTables ...doltdb.TableName,
 ) (doltdb.RootValue, error) {
-	if len(tables) == 0 {
+	if len(originalTables) == 0 {
 		return root, nil
 	}
 
-	// TODO: support multiple schemas in the same set
-	tableMap, err := root.getTableMap(ctx, tables[0].Schema)
-	if err != nil {
-		return nil, err
+	var tables []doltdb.TableName
+	var rootObjNames []struct {
+		rawID id.Id
+		objID rootobject.RootObjectID
 	}
-
-	edits := make([]tableEdit, len(tables))
-	for i, name := range tables {
-		a, err := tableMap.Get(ctx, name.Name)
+	for _, name := range originalTables {
+		_, rawID, objID, err := rootobject.ResolveName(ctx, root, name)
```

Comment:
Same comment. This is going to be dog slow compared to checking for tables first (the common case)